docs/wave/workgroup_reordering.rst

```diff
@@ -0,0 +1,176 @@
+
+Default vs Reordered Launch Grids
+=================================
+
+Let's look at the column-major workgroup ordering for the same 8x8 output matrix from before.
+
+.. image:: ./default_wg_reads.png
+   :alt: Data reads with default workgroups
+   :align: center
+
+We can see to calculate 8 output blocks for C, we have to read all 64 blocks of data from A and 8 blocks of data from B for a total of 72 blocks.
+
+Instead of this default column-major workgroup order, we can apply a custom workgroup reordering strategy:
+
+.. image:: ./reordered_wg_reads.png
+   :alt: Data reads with reordered workgroups
+   :align: center
+
+Here, we group 4 workgroups along the N dimension of the data (across 4 columns) before moving along the M dimension of the data (moving down a row) and assigning the next set of workgroups.
+4 in this case can be considered the grouping factor, and after we have fully assigned all 4 columns to workgroups, we can start assigning workgroups to the next 4 columns in the same manner.
+With this reordering, to compute 8 output blocks, we only need to read 48 total input blocks as opposed to 72 from before — reducing redundant memory reads and improving cache utilization.
+
+Reordering Logic: Code Example
+==============================
+
+We can implement this reordering scheme as follows:
+
+.. code-block:: python
+
+    wg0, wg1 = WORKGROUP_0, WORKGROUP_1
+    num_wg_0 = ceiling(M / BLOCK_M)
+
+    # Flatten the workgroup index
+    flat_wg_index = wg1 * num_wg_0 + wg0
+
+    # Define group size
+    num_wg_group = GROUP_SIZE_N * num_wg_0
+    group_id = flat_wg_index // num_wg_group
+    first_wg_id_1 = group_id * GROUP_SIZE_N
+
+    # Compute new reordered coordinates
+    new_wg0 = (flat_wg_index % num_wg_group) // GROUP_SIZE_N
+    new_wg1 = first_wg_id_1 + (flat_wg_index % num_wg_group) % GROUP_SIZE_N
+
+    # Apply constraints
+    constraints += [tkw.ReorderingConstraint(new_wg0, 0)]
+    constraints += [tkw.ReorderingConstraint(new_wg1, 1)]
+
+Explanation
+===========
+
+Here, `wg0` and `wg1` represent the original workgroup coordinates. We:
+
+1. Flatten the 2D index into a 1D linear index (`flat_wg_index`)
+2. Define how many workgroups make up one "group"
+3. Compute a new pair `(new_wg0, new_wg1)` representing the reordered coordinates
+
+The `GROUP_SIZE_N` parameter is our grouping factor, and it controls how many columns of workgroups we pack together before moving to the next row of groups.
+
+These new coordinates are then fed into `tkw.ReorderingConstraint`, which updates the launch grid accordingly, behind the scenes.
+
+Results
+=======
+
+This section compares performance between normal and reordered workgroup scheduling across matrix shapes. Two key metrics are analyzed:
+
+1. **L2 Cache Hit Rate**
+2. **Kernel Execution Time**
+
+L2 Cache Hit Rate (%)
+------------------
+
++-------------------------+----------------+------------------+
```

Comment:
Can you add the tile size and scheduletype.prefetch here S.T people can repro this